.ci/generate-releases.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#
+# It produces the list of current releases, BCs or upcoming ones using the artifacts-api and GitHub API.
+# Then it does some manipulation to discard those releases that are not available. A bit opinionated.
+#
+set -eo pipefail
+
+
+###############
+### FUNCTIONS
+###############
+
+# Get the latest github release for the given tag prefix.
+# Releases starts with v<major>, i.e: v8
+# It uses the gh cli in elastic/elasticsearch.
+# Owned by: release team
+function latest() {
+  local version="${1}"
+  gh api repos/elastic/elasticsearch/releases \
```

Comment:
nit: for commands like this (or `docker pull`) that access network resources, it's preferable to retry them in case of failure. We have https://github.com/minfrin/retry preinstalled on our custom VM images and containers, but there are also bash retry helpers like here: https://github.com/elastic/fleet-server/blob/f099b91b62043fd093cea1f03dfac74d5d5767b9/.buildkite/scripts/common.sh#L62-L78

Reply:
https://github.com/elastic/apm-pipeline-library/pull/2479/commits/42bfe48a0e75dc250c3a1968d1f5943b630d0743 uses the retry in all the commands interacting with third-party services. It does not print any warning messages to work nicely with the existing functions.